torchmdnet/priors.py

```diff
@@ -76,3 +79,52 @@ def get_init_args(self):
 
     def forward(self, x, z, pos, batch):
         return x + self.atomref(z)
+
+
+class ZBL(BasePrior):
+    """This class implements the Ziegler-Biersack-Littmark (ZBL) potential for screened nuclear repulsion.
+    Is is described in https://doi.org/10.1007/978-3-642-68779-2_5 (equations 9 and 10 on page 147).  It
+    is an empirical potential that does a good job of describing the repulsion between atoms at very short
+    distances.
+
+    To use this prior, the Dataset must provide the following attributes.
+
+    atomic_number: 1D tensor of length max_z.  atomic_number[z] is the atomic number of atoms with atom type z.
+    distance_scale: multiply by this factor to convert coordinates stored in the dataset to meters
```

Comment:
I went with SI units to avoid preferencing any particular choice of units.  For what it's worth, eV is rarely used in MD simulations.  Most MD codes use either kJ/mol or kcal/mol as their energy unit.

Reply:
Can't you do the opposite? I.e. specify in the ZBL prior what units are used by the NNP?

Reply:
This way everyone can keep using their units and ZBL will adapt to it.

Reply:
The neural network is agnostic with regard to units.  It receives inputs and tries to make its outputs match the ones in the dataset.  It doesn't know or care what units they're in.

A physics based potential does depend on units.  It needs to know what units the coordinates are in, and it needs to know what units the energy is expected to be in.